--- /home/armol/ulefone-normal-v5-policy-test/v3.rules	2026-08-29 20:44:47.622404972 +0300
+++ /home/armol/ulefone-normal-v5-policy-test/v5.rules	2026-08-29 20:44:48.934404861 +0300
@@ -25416,6 +25416,7 @@
 allow kernel kernel:lockdown confidentiality;
 allow kernel kernel:process { fork getattr getcap getpgid getsched getsession setcap setpgid setrlimit setsched sigchld sigkill signal signull sigstop };
 allow kernel kernel:security setcheckreqprot;
+allow kernel kernel:system syslog_read;
 allow kernel kernel:unix_dgram_socket { append bind connect create getattr getopt ioctl lock map read sendto setattr setopt shutdown write };
 allow kernel kernel:unix_stream_socket { accept append bind connect connectto create getattr getopt ioctl listen lock map read setattr setopt shutdown write };
 allow kernel kmsg_device:chr_file { getattr open read relabelto write };
